packages/taco/integration-test/encrypt-decrypt.test.ts

```diff
@@ -128,5 +131,66 @@ describe.skipIf(!process.env.RUNNING_IN_CI)(
       // Verify decryption
       expect(decryptedMessageString).toEqual(messageString);
     }, 15000); // 15s timeout
+
+    // TODO: Unskip this test when wallet allowlist condition is implemented and running at the nodes as per https://github.com/nucypher/nucypher/issues/3600
+    test.skip('should encrypt and decrypt according to wallet allowlist condition', async () => {
+      // Create test message
+      const messageString =
+        'This message should only be accessible to allowed wallet addresses';
+      const message = toBytes(messageString);
+
+      // Create wallet allowlist condition with consumer address in the list
+      const addressAllowlistCondition =
+        new conditions.base.addressAllowlist.AddressAllowlistCondition({
+          userAddress: ':userAddress',
+          addresses: [
+            CONSUMER_ADDRESS,
+            '0x742d35Cc6634C0532925a3b844Bc454e4438f44e', // Some other address
+            '0x0000000000000000000000000000000000000001', // Another address
+          ],
+        });
+
+      // Verify that the condition requires authentication
+      expect(addressAllowlistCondition.requiresAuthentication()).toBe(true);
+
+      // Encrypt message with the wallet allowlist condition
+      const messageKit = await encrypt(
+        provider,
+        DOMAIN,
+        message,
+        addressAllowlistCondition,
+        RITUAL_ID,
+        encryptorSigner,
+      );
+
+      const encryptedBytes = messageKit.toBytes();
+
+      // Prepare for decryption
+      const messageKitFromBytes = ThresholdMessageKit.fromBytes(encryptedBytes);
+      const conditionContext =
+        conditions.context.ConditionContext.fromMessageKit(messageKitFromBytes);
+
+      // Add auth provider for the consumer wallet
+      const authProvider = new EIP4361AuthProvider(provider, consumerSigner, {
+        domain: 'localhost',
+        uri: 'http://localhost:3000',
```

Comment:
It's fine to provide these values, but just FYI you don't need to provide them since there are defaults that are used when the values are not provided.